File: 1-99/9/9.cpp
using namespace std;

class Solution {
private:
	// Reverse digits of an int, it may overflow as int, so return as long
	long reverseInt(int x) {
		long rev = 0;
		for (int cpy = x; cpy > 0; cpy /= 10)
			rev = rev * 10 + cpy % 10;
		return rev;
	}
public:
	bool isPalindrome(int x) {
		if (x < 0 || (x % 10 == 0 && x != 0))
			// All negative ints are not palindrome
			return false;
		else if (x < 10)
			// All single digit positive ints are palindrome
			return true;

		return x == reverseInt(x);
	}
};

File: 0001-0099/9/9.go
func reverseInt(x int) int64 {
	var rev int64 = 0
	for cpy := int64(x); cpy > 0; cpy /= 10 {
		rev = rev * 10 + cpy % 10;
	}
	return rev
}

func isPalindrome(x int) bool {
	if (x < 0 || (x % 10 == 0 && x != 0)) {
		return false
	} else if (x < 10) {
		return true
	}

	return x == int(reverseInt(x))
}
